### PR TITLE
[Storage] [DataMovement] Added support for Share File to/from Blob for Single Transfer

### PR DESCRIPTION
**Pull request by amnguye**
Mainly this was ensuring that the Single Transfer Share File to/from Blob transfers maintained properties that were preservable/settable. Which just contained a lot of test updates.

One change that had to be made was ensuring that Content Language and Content Encoding property type was converted properly from object to string array to string. Since Blobs accepts these as a `string` while Shares accept this value as a `string[]`

### CODE REVIEW

File: sdk/storage/Azure.Storage.DataMovement.Blobs/src/DataMovementBlobsExtensions.cs [hunk 1]
```diff
@@ -637,34 +638,50 @@ internal static StorageResourceItemProperties ToResourceProperties(this BlobItem
...
+            {
+                return Convert.ToString(contentPropertyValue);
+            }
+            else if (contentPropertyValue is string[])
+            {
+                return string.Join(",", (string[])contentPropertyValue);
+            }
+            else
+            {
+                throw Errors.UnexpectedPropertyType(DataMovementConstants.ResourceProperties.ContentType, DataMovementConstants.StringTypeStr, DataMovementConstants.StringArrayTypeStr);
```
Comment (jalauzon-msft) [inline sdk/storage/Azure.Storage.DataMovement.Blobs/src/DataMovementBlobsExtensions.cs:653]: This isn't always `DataMovementConstants.ResourceProperties.ContentType` right? I guess we could take this in?

File: sdk/storage/Azure.Storage.DataMovement.Blobs/src/DataMovementBlobsExtensions.cs [hunk 2]
```diff
@@ -637,34 +638,50 @@ internal static StorageResourceItemProperties ToResourceProperties(this BlobItem
...
+
         private static BlobHttpHeaders GetHttpHeaders(
             BlobStorageResourceOptions options,
             Dictionary<string, object> properties)
             => new()
             {
                 ContentType = (options?.ContentType?.Preserve ?? true)
                     ? properties?.TryGetValue(DataMovementConstants.ResourceProperties.ContentType, out object contentType) == true
-                        ? (string) contentType
+                        ? (contentType is string) ? Convert.ToString(contentType) : throw Errors.UnexpectedPropertyType(DataMovementConstants.ResourceProperties.ContentType, DataMovementConstants.StringTypeStr)
```
Comment (jalauzon-msft) [inline sdk/storage/Azure.Storage.DataMovement.Blobs/src/DataMovementBlobsExtensions.cs:664]: do we just want to call `ConvertContentPropertyObjectToString` for all these? It should do the same thing?

**Review by jalauzon-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.DataMovement.Blobs/src/DataMovementBlobsExtensions.cs [hunk 3]
```diff
@@ -637,34 +638,50 @@ internal static StorageResourceItemProperties ToResourceProperties(this BlobItem
                 properties: properties);
         }
 
+        private static string ConvertContentPropertyObjectToString(object contentPropertyValue)
+        {
+            if (contentPropertyValue is string)
+            {
+                return Convert.ToString(contentPropertyValue);
```
[the remark this answers is not in the document]
Comment (amnguye) [reply sdk/storage/Azure.Storage.DataMovement.Blobs/src/DataMovementBlobsExtensions.cs]: done

File: sdk/storage/Azure.Storage.DataMovement.Blobs/src/DataMovementBlobsExtensions.cs (hunk 2 above)
Comment (amnguye) [reply sdk/storage/Azure.Storage.DataMovement.Blobs/src/DataMovementBlobsExtensions.cs]: reverted back.

**Review by jaschrep-msft — APPROVED** [review]

**Pull request merged**